### PR TITLE
Feature/vmk 2658 provide supplementary view support in vmk collection view data source

### PR DESCRIPTION
**Pull request by peterdarbey**
Added VMK support for SupplementaryView in UICollectionView with associated optional VM.

### CODE REVIEW

File: ViewModelKit/UICollectionReusableView/VMKCollectionReusableHeaderView.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
+//
+//  VMKCollectionReusableHeaderView.h
+//  ViewModelKit
+//
+//  Created by Peter Darbey on 08/06/2020.
+//  Copyright © 2020 Andre Trettin. All rights reserved.
```
Comment (Andrenalin) [inline ViewModelKit/UICollectionReusableView/VMKCollectionReusableHeaderView.h:6]: Maybe not really true to put my name here.....

File: ViewModelKit/UICollectionReusableView/VMKCollectionReusableHeaderView.h [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+//
+
+@import UIKit;
+
+#import "VMKViewHeaderFooterType.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface VMKCollectionReusableHeaderView<__covariant HeaderViewModel:__kindof VMKViewModel<VMKHeaderFooterType> *> : UICollectionReusableView <VMKViewHeaderFooterType>
```
Comment (Andrenalin) [inline ViewModelKit/UICollectionReusableView/VMKCollectionReusableHeaderView.h:15]: nice annotations 👍

**Review by Andrenalin — COMMENTED** [review]
From the first glance it looks good.
Are you planning to write a UT for the new feature?

File: ViewModelKit/UICollectionReusableView/VMKCollectionReusableHeaderView.h (hunk 1 above)
Comment (peterdarbey) [reply ViewModelKit/UICollectionReusableView/VMKCollectionReusableHeaderView.h:6]: I guess this was auto generated

**Pull request closed without merging**